packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -359,8 +359,8 @@ class AppDomain extends Domain {
       connectionInfoCompleter = new Completer<DebugConnectionInfo>();
       connectionInfoCompleter.future.then((DebugConnectionInfo info) {
         Map<String, dynamic> params = <String, dynamic>{
-          'port': info.port,
-          'wsUri': info.wsUri
+          'port': info.httpUri.port,
```

Comment:
I forget if this is for the future or CL or not but you only need to pass 1 Uri here. Everything else (scheme differences, ports) can be derived on the other end.